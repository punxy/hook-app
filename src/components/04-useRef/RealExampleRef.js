import React, { useState } from 'react';
import { MultipleCustomHooks } from '../03-example/MultipleCustomHooks';
import '../effects.css';


export const RealExampleRef = () => {

    const [show, setShow] = useState(false);

    return (
        <div>
            <h1>Real Example Ref</h1>
            <hr />

            { show && <MultipleCustomHooks /> }


            <button className='btn tbn-primary' onClick={() => setShow(!show)}>show/hide</button>
            
        </div>
    )
}
