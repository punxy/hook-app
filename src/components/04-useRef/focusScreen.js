import React, { useRef } from 'react';
import '../effects.css';

export const FocusScreen = () => {

    const inputRef = useRef()

    const handleClick = () => {
        //document.querySelector('input').focus();
        inputRef.current.focus();
    }

    
    return (
         <>
            <h1>Focus Screen</h1>
            <hr />

            <input 
                ref={inputRef}
                className='form-control' 
                placeholder='Su nombre'
            />

            <button 
                className='btn btn-outline-primary mt-5'
                onClick={handleClick}
            >
                Focus
            </button>
        </>
    )
}
