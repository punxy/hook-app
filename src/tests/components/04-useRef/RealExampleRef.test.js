import React from 'react';
import { shallow } from "enzyme";
import '@testing-library/jest-dom';
import { RealExampleRef } from '../../../components/04-useRef/RealExampleRef';

describe('Test in RealExampleRef component', () => {

    test('should be render correctly', () => {
        const wrapper = shallow( <RealExampleRef /> );
        expect(wrapper).toMatchSnapshot();
        expect( wrapper.find('MultipleCustomHooks').exists() ).toBeFalsy();
    });

    test('should be render a MultipleCustomHook component', () => {
        const wrapper = shallow( <RealExampleRef /> );
        wrapper.find('button').simulate('click');
        expect( wrapper.find('MultipleCustomHooks').exists() ).toBeTruthy();
    });
});