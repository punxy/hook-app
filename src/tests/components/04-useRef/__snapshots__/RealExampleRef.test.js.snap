// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in RealExampleRef component should be render correctly 1`] = `
<div>
  <h1>
    Real Example Ref
  </h1>
  <hr />
  <button
    className="btn tbn-primary"
    onClick={[Function]}
  >
    show/hide
  </button>
</div>
`;
